Whisky/Views/Bottle/Pins/PinCreationView.swift

```diff
@@ -0,0 +1,134 @@
+                        nameValid = !name.isEmpty
+                    }
+            }
+            HStack {
+                Text("pin.path")
+                Spacer()
+                Button("create.browse") {
+                    let panel = NSOpenPanel()
+                    panel.canChooseFiles = true
+                    panel.allowedContentTypes = [UTType.exe,
+                                                 UTType(exportedAs: "com.microsoft.msi-installer"),
+                                                 UTType(exportedAs: "com.microsoft.bat")]
+                    panel.directoryURL = bottle.url.appending(path: "drive_c")
+                    panel.canChooseDirectories = false
+                    panel.allowsMultipleSelection = false
+                    panel.canCreateDirectories = false
+                    panel.begin { result in
+                        if result == .OK {
+                            if let url = panel.urls.first {
+                                newPinURL = url
+                            }
+                        }
+                    }
+                }
+            }
+            HStack {
+                Text(pinPath)
+                    .truncationMode(.middle)
+                    .lineLimit(2)
+                    .help(pinPath)
+            }
+            Spacer()
+            HStack {
+                Spacer()
+                Button("create.cancel") {
+                    dismiss()
+                }
+                .keyboardShortcut(.cancelAction)
+                Button("pin.create") {
+                    let newlyCreatedPin = Program(name: newPinName, url: newPinURL, bottle: bottle)
+                    let existingProgram = bottle.programs.first(where: { program in
+                        program.url == newlyCreatedPin.url
+                    })
+                    // Ensure this URL isn't already pinned
+                    errorMessage = String(format: String(localized: "pin.error.duplicate"),
+                                          newPinURL.lastPathComponent)
+                    didError = existingProgram != nil && existingProgram?.pinned ?? false
+                    if !didError {
+                        // If this is a new program, add it to the array
+                        if existingProgram != nil {
+                            bottle.programs.append(newlyCreatedPin)
+                        }
+                        newlyCreatedPin.pinned = true
+                        // Trigger a reload
+                        bottle.settings.pins = bottle.settings.pins
+                        bottle.updateInstalledPrograms()
+                        dismiss()
+                    }
+                }
+                .keyboardShortcut(.defaultAction)
+                .disabled(!nameValid)
+                .alert(String(localized: "pin.error.title"),
+                    isPresented: $didError
+                ) {
+                    Button("button.ok") {
+                    }
```

Comment:
I believe you can get rid of this button, the alert will add its own system button

Reply:
You're right; this was an unfortunate copy/paste result. Good catch.